eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -348,4 +348,52 @@ class GraphSpec extends AnyFunSuite {
     assert(paths(1).path == Seq(edgeCE, edgeEG, edgeGH))
     assert(paths(2).path == Seq(edgeCD, edgeDF, edgeFH))
   }
+
+  test("RoutingHeuristics.normalize when value in range") {
+    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 0, max = 100) == 0.100008)
+    assert(Graph.RoutingHeuristics.normalize(value = 20, min = 10, max = 200) == 0.05264052631578948)
+    assert(Graph.RoutingHeuristics.normalize(value = -11, min = -100, max = -10) == 0.9888791111111112)
+  }
+
+  test("Graph.RoutingHeuristics.normalize when value is == min") {
+    assert(Graph.RoutingHeuristics.normalize(value = 0, min = 0, max = 100) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 10, min = 10, max = 200) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = -100, min = -100, max = -10) == 1.0E-5)
+  }
+
+  test("Graph.RoutingHeuristics.normalize when value is < min") {
+    assert(Graph.RoutingHeuristics.normalize(value = -1, min = 0, max = 100) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = 9.1, min = 10, max = 200) == 1.0E-5)
+    assert(Graph.RoutingHeuristics.normalize(value = -101.1, min = -100, max = -10) == 1.0E-5)
+  }
+
+  test("Graph.RoutingHeuristics.normalize when value is == max") {
+    assert(Graph.RoutingHeuristics.normalize(value = 100, min = 0, max = 100) == 0.99999)
+    assert(Graph.RoutingHeuristics.normalize(value = 200, min = 10, max = 200) == 0.9999899999999999)
```

Comment:
How come you don't expect the same result in both of these cases? It looks like a bug, but it's actually not because we don't care about the exact value.